aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,327 @@
+
+To see the scope-based ordering in action, consider the following class, which has the same filter applied at both the class and method level. The same filter is also registered globally. When run, the log output will show the order in which the filters executed, and how these compared to the ``Controller`` base class's ``OnActionExecuting`` and ``OnActionExecuted`` methods.
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/OrderingController.cs
+  :language: c#
+  :emphasize-lines: 1,10,18,24
+  :lines: 8-34
+  :dedent: 4
+
+The resulting order is:
+	#. The Controller ``OnActionExecuting``
+	#. The Global filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuting``
+	#. The Method filter ``OnActionExecuting``
+	#. The Method filter ``OnActionExecuted``
+	#. The Class filter ``OnActionExecuted``
+	#. The Global filter ``OnActionExecuted``
+	#. The Controller ``OnActionExecuted``
+
+.. note:: ``IOrderedFilter`` trumps scope when determining the order in which filters will run. Filters are sorted first by order, then scope is used to break ties. Order defaults to 0 if not set.
+
+To modify the default, scope-based order, you could explicitly set the ``Order`` property of the class-level and/or method-level filters. For example, adding ``Order=-1`` to the action method:
+
+.. code-block:: c#
+
+	[OrderLoggingActionFilter(Name = "Method Level Attribute", Order=-1)]
+
+In this case, a value of less than zero would ensure this filter ran before both the Global and Class level filters.
+
+The new order would be:
+	#. The Controller ``OnActionExecuting``
+	#. The Method filter ``OnActionExecuting``
+	#. The Global filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuting``
+	#. The Class filter ``OnActionExecuted``
+	#. The Global filter ``OnActionExecuted``
+	#. The Method filter ``OnActionExecuted``
+	#. The Controller ``OnActionExecuted``
+
+.. note:: The ``Controller`` class's methods always run before and after all filters. These methods are not implemented as ``IFilter`` instances and do not participate in the ``IFilter`` ordering algorithm.
+
+.. _authorization-filters:
+
+Authorization Filters
+---------------------
+*Authorization Filters* control access to action methods, and are the first filters to be executed within the filter pipeline. They have only a before stage, unlike most filters that support before and after methods. You should only write a custom authorization filter if you are writing your own authorization framework. Note that you should not throw exceptions within authorization filters, since nothing will handle the exception (exception filters won't handle them). Instead, issue a challenge or find another way. They are covered in the :doc:`Security </security/index>` section of the documentation.
+
+Learn more about :doc:`/security/authorization/authorization-filters`.
```

Comment:
I think the authorization filters article will go away as the functionality of the filter is covered in the various articles under https://docs.asp.net/en/latest/security/authorization/index.html, so lets point to that instead.